apps/prairielearn/assets/scripts/lib/parallelHistograms.ts

```diff
@@ -0,0 +1,201 @@
+import { scaleLinear, scaleBand, axisBottom, axisLeft, range, select } from 'd3';
+
+interface Data {
+  label: string;
+  histogram: number[];
+  mean: number;
+}
+
+export function parallel_histograms(
+  selector: Element,
+  data: Data[],
+  options: {
+    width?: number;
+    height?: number;
+    xTickLabels?: string[] | 'auto';
+    yTickLabels?: string[] | 'auto';
+    xAxisHeight?: number;
+    yAxisWidth?: number;
+    xgrid?: number[];
+    ygrid?: number[];
+    xlabel?: string;
+    ylabel?: string;
+    topPadding?: number;
+    rightPadding?: number;
+  },
+) {
+  const resolvedOptions = {
+    width: 600,
+    height: 370,
+    xTickLabels: 'auto',
+    yTickLabels: 'auto',
+    yAxisWidth: 70,
+    xAxisHeight: 70,
+    topPadding: 15,
+    rightPadding: 2,
+    ...options,
+  };
+
+  const width: number = resolvedOptions.width ?? 600;
```

Comment:
Is the type annotation here really necessary?

Reply:
No, removed in [bd59c6c](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/bd59c6c63f9729486ff37c2971ec617424f9ce43).